src/azure-cli/azure/cli/command_modules/appconfig/feature.py

```diff
@@ -142,17 +147,28 @@ def set_feature(cmd,
 
 
 def delete_feature(cmd,
-                   feature,
+                   feature=None,
+                   key=None,
                    name=None,
                    label=None,
                    yes=False,
                    connection_string=None,
                    auth_mode="key",
                    endpoint=None):
+    if key is None and feature is None:
+        raise CLIError("Please provide either `--key` or `--feature` value.")
+    if key and feature:
+        logger.warning("Since both `--key` and `--feature` are provided, `--feature` argument will be ignored.")
+
+    if key is not None:
+        feature_pattern = key[len(FeatureFlagConstants.FEATURE_FLAG_PREFIX):]
+    elif feature is not None:
+        feature_pattern = feature
+
     azconfig_client = get_appconfig_data_client(cmd, name, connection_string, auth_mode, endpoint)
 
     retrieved_keyvalues = __list_all_keyvalues(azconfig_client,
```

Comment:
>__list_all_keyvalues [](http://example.com/codeflow?start=26&length=20)

Since we introduce the appconfig key, why don't we list items by key directly. Each caller should decide how to get the key.

Reply:
Done.